### PR TITLE
[xcode9] Bind NSXpcConnection and related classes

### PR DESCRIPTION
**Pull request by timrisi**
Needs a sample ported to make sure it all works correctly before merging.

https://trello.com/c/gl98lkAu/26-port-sandboxedfetch-sample-to-make-sure-nsxpcconnection-bindings-work

### CODE REVIEW

File: src/Foundation/Enum.cs [hunk 1]
```diff
@@ -1258,4 +1258,9 @@ public enum NSMeasurementFormatterUnitOptions : nuint {
 		TemperatureWithoutUnit = (1 << 2)
 	}
 
+	[Watch (2, 0), TV (9, 0), Mac (10, 8), iOS (6, 0)]
+	[Native]
+	public enum NSXpcConnectionOptions : nuint {
+		Privileged = (1 << 12)
```
Comment (dalexsoto) [inline src/Foundation/Enum.cs]: Minor: comma at the end.

File: src/foundation.cs [hunk 2]
```diff
@@ -14274,4 +14274,168 @@ interface NSFileProviderMessageInterface : NSSecureCoding
 		[Export ("name")]
         	string Name { get; }
 	}
+
+	interface INSXpcProxyCreating
+	{
+	}
+
+	[Protocol]
+	[BaseType (typeof(NSObject), Name="NSXPCProxyCreating")]
```
Comment (spouliot) [inline src/foundation.cs]: a protocol should not have a `[BaseType]` unless a `[Model]` is required

File: src/Foundation/Enum.cs (hunk 1 above)
Comment (chamons) [inline src/Foundation/Enum.cs]: Flags?

**Comment by chamons** [comment]
I agree, this needs a sample before I'd be really happy with it.

File: src/foundation.cs [hunk 3]
```diff
@@ -14274,4 +14274,168 @@ interface NSFileProviderMessageInterface : NSSecureCoding
...
+	}
+
+	[Mac (10,8)]
+	[BaseType (typeof(NSObject), Name="NSXPCInterface")]
+	[DisableDefaultCtor]
+	interface NSXpcInterface
+	{
+		[Static]
+		[Export ("interfaceWithProtocol:")]
+		NSXpcInterface FromInterface (Protocol protocol);
```
Comment (spouliot) [inline src/foundation.cs]: bad name, it's not _from an interface_ but creates an interface from a protocol, it should be `FromProtocol`

File: src/foundation.cs [hunk 4]
```diff
@@ -14274,4 +14274,168 @@ interface NSFileProviderMessageInterface : NSSecureCoding
...
+
+		[Export ("protocol", ArgumentSemantic.Assign)]
+		Protocol Protocol { get; set; }
+
+		[Export ("setClasses:forSelector:argumentIndex:ofReply:")]
+		void SetClasses (NSSet<Class> classes, Selector sel, nuint arg, bool ofReply);
```
Comment (spouliot) [inline src/foundation.cs]: is not ideal to expose (only) API with `Selector` as we do our best to hide them

**Comment by timrisi** [comment]
I don't know of anything it's blocking, just saw the new bindings from Xcode 9 and it hadn't been previously bound at all, so this was adding in the original bindings along with the new ones

**Comment by chamons** [comment]
I saw it was part of the generic foundation PR and requested it be split out since it's (likely) blocked by sample.

**Comment by rolfbjarne** [comment]
Tests need to verify that none of the API returns an NSProxy object (because NSProxy doesn't inherit from NSObject, and that breaks assumptions on our runtime).

**Comment by mandel-macaque** [comment]
Closing as PR was abandoned. Created an issue to follow up the work: https://github.com/xamarin/xamarin-macios/issues/7488

**Pull request closed without merging**